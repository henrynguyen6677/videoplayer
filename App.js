import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity
} from 'react-native';

import { Video } from 'expo-av';

class PlaylistItem {
  constructor(key, name, uri, isVideo) {
    this.key = key;
    this.name = name;
    this.uri = uri;
  }
}

const PLAYLIST = [
  new PlaylistItem(
    '0',
    'Big Buck Bunny',
    'http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4',
  ),
  new PlaylistItem(
    '1',
    "Popeye - I don't scare",
    'https://ia800501.us.archive.org/11/items/popeye_i_dont_scare/popeye_i_dont_scare_512kb.mp4',
  ),
  new PlaylistItem(
    '2',
    'Big Buck Bunny',
    'http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4',
  ),
  new PlaylistItem(
    '3',
    "Popeye - I don't scare",
    'https://ia800501.us.archive.org/11/items/popeye_i_dont_scare/popeye_i_dont_scare_512kb.mp4',
  ),
  new PlaylistItem(
    '5',
    'Big Buck Bunny',
    'http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4',
  ),
  new PlaylistItem(
    '6',
    "Popeye - I don't scare",
    'https://ia800501.us.archive.org/11/items/popeye_i_dont_scare/popeye_i_dont_scare_512kb.mp4',
  ),
];

export default class App extends Component {
  constructor(props) {
    super(props);
    this.index = 0;
    this.state = {
      source: {
        uri: PLAYLIST[0].uri
      }
    };
  }
  
  _onPressButton = (item) => {
    console.log('[_onPressButton]');
    this.setState((state, props) => {
      return { source: { uri: item.uri }};
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Video
          isLoaded
          rate={1.0}
          volume={1.0}
          isMuted={false}
          resizeMode="cover"
          shouldPlay
          isLooping      
          source={this.state.source}
          style={ styles.video }
        />
        <FlatList
          data={ PLAYLIST }
          renderItem={ ({item}) =>
            <View key={ item.key }>
              <TouchableOpacity
                onPress={ () => this._onPressButton(item)}
              >
                <Text style={styles.item}>
                  {item.name}
                </Text>
                
              </TouchableOpacity>
            </View>
          }
        >
        </FlatList>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
    alignSelf: 'stretch',
  },
  video: {
    width: 300,
    height: 300,
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});
