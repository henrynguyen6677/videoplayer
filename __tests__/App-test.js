import React from 'react';
import App from '../App.js';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import {
  FlatList
} from 'react-native';
import { Video } from 'expo-av';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('shows a Video', () => {
  let wrapped = shallow(<App />);  
  expect(wrapped.find(Video).length).toEqual(1);
});

it('shows a Video list', () => {
  let wrapped = shallow(<App />);  
  expect(wrapped.find(FlatList).length).toEqual(1);
});

