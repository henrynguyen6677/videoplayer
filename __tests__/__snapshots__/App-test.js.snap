// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginTop": 20,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "height": 300,
        "overflow": "hidden",
        "width": 300,
      }
    }
  >
    <ViewManagerAdapter_ExpoVideoView
      proxiedProperties={
        Object {
          "isLoaded": true,
          "onError": [Function],
          "onFullscreenUpdate": [Function],
          "onLoad": [Function],
          "onLoadStart": [Function],
          "onReadyForDisplay": [Function],
          "onStatusUpdate": [Function],
          "resizeMode": "mock",
          "source": Object {
            "headers": undefined,
            "overridingExtension": null,
            "uri": "http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4",
          },
          "status": Object {
            "isLooping": true,
            "isMuted": false,
            "rate": 1,
            "shouldPlay": true,
            "volume": 1,
          },
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <RCTScrollView
    data={
      Array [
        PlaylistItem {
          "key": "1",
          "name": "Big Buck Bunny",
          "uri": "http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4",
        },
        PlaylistItem {
          "key": "2",
          "name": "Popeye - I don't scare",
          "uri": "https://ia800501.us.archive.org/11/items/popeye_i_dont_scare/popeye_i_dont_scare_512kb.mp4",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "height": 44,
                  "padding": 10,
                }
              }
            >
              Big Buck Bunny
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "height": 44,
                  "padding": 10,
                }
              }
            >
              Popeye - I don't scare
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
